common/common.go

```diff
@@ -412,3 +430,66 @@ func dropInvalidCPUs(nums []int, maxcpu int) []int {
 	}
 	return nums[:i]
 }
+// parse ip address string to int ip
+func StringToIPv4(ipaddr string) uint32 {
+	str_ary := strings.Split(ipaddr, ".")
+	bIp := [4]byte{}
+
+	i := 0
+	for _, element := range str_ary {
+		val, err := strconv.Atoi(element)
+		if err != nil {
+			panic(err)
+		}
+		bIp[i] = byte(val)
+		i++
+	}
+	return ByteAryToIPv4(bIp)
+
+}
+
+//get the next ip address specified by the inc
+func NextIP(ip net.IP, inc uint) net.IP {
+	i := ip.To4()
+	v := uint(i[0])<<24 + uint(i[1])<<16 + uint(i[2])<<8 + uint(i[3])
+	v += inc
+	v3 := byte(v & 0xFF)
+	v2 := byte((v >> 8) & 0xFF)
+	v1 := byte((v >> 16) & 0xFF)
+	v0 := byte((v >> 24) & 0xFF)
+	return net.IPv4(v0, v1, v2, v3)
+}
+
+//increment ip address
+func IncrementIP(ip net.IP) {
+	for j := len(ip) - 1; j >= 0; j-- {
+		ip[j]++
+		if ip[j] > 0 {
+			break
+		}
+	}
+}
+// convert ip address to int
+func Ip2int(ip net.IP) uint32 {
+	if len(ip) == 16 {
+		return binary.BigEndian.Uint32(ip[12:16])
+	}
+	return binary.BigEndian.Uint32(ip)
+}
+
+//convert int ip to  net.IP
+func Int2ip(nn uint32) net.IP {
+	ip := make(net.IP, 4)
+	binary.BigEndian.PutUint32(ip, nn)
+	return ip
+}
+
+// byte arr to ipv4 int
+func ByteAryToIPv4(ipB [4]byte) uint32 {
```

Comment:
We have this function in packet/packet.go

Reply:
ok